### ISSUE TITLE
Got mAP of 62.81

### ISSUE THREAD

**Issue by JesseYang** [marvis/pytorch-yolo2#5]
I follow the ReadMe to train with VOC 07+12 and evaluate on VOC 2007 test set. The master branch is used. The mAP is 62.81. Do I miss anything?

**Comment by marvis** [comment]
Yes, it is a problem to reproduce the original results. I am always trying to figure out the problem right now.  You can check region_loss implementation on https://github.com/marvis/pytorch-yolo2/blob/dev/region_loss.py, which is exactly the same with darknet's implementation. But still a problem to get the same results. 

And I suggest to divide all the loss below by nGT
---------
loss_x = self.coord_scale * nn.MSELoss(size_average=False)(x*coord_mask, tx*coord_mask)/nGT
        loss_y = self.coord_scale * nn.MSELoss(size_average=False)(y*coord_mask, ty*coord_mask)/nGT
        loss_w = self.coord_scale * nn.MSELoss(size_average=False)(w*coord_mask, tw*coord_mask)/nGT
        loss_h = self.coord_scale * nn.MSELoss(size_average=False)(h*coord_mask, th*coord_mask)/nGT
        loss_conf = nn.MSELoss(size_average=False)(conf*conf_mask, tconf*conf_mask)/nGT
        loss_cls = self.class_scale * nn.CrossEntropyLoss(size_average=False)(cls*cls_mask, tcls)/nGT

which may get better results.

**Comment by JesseYang** [comment]
@marvis Do you mean that you previously achieved the mAP claimed in the RADME.md, but cannot reproduce it?

**Comment by marvis** [comment]
sorry for the misunderstanding, the mAP is for yolo-voc.weights from darknet. I will make it more clear.

**Comment by JesseYang** [comment]
I see. Thanks!

**Comment by marvis** [comment]
@JesseYang  You can try the most recently code now. Mean AP = 0.7293

**Comment by marvis** [comment]
@JesseYang  You can try the most recent code now. Mean AP = 0.7293

**Comment by njtuzzy** [comment]
@marvis may i ask what did you do to imporve the mAP to 0.7293?

**Comment by marvis** [comment]
the warm up training is very important.

**Comment by wwxJream** [comment]
@marvis Hi~ I only get 59mAP when use yolo-voc.cfg to train for 200 epochs. What should I do to get the 72mAP. I only change the batch_size to 16 and learning rate to 0.00025. Please help me~ Thank you very much!!

**Comment by txdat** [comment]
i have problem that bbox_loss (or tx, ty, h, w loss) not learning after warmup, results are misclassification and bbox's shape are same with anchors, what happened? thanks

**Comment by abhigoku10** [comment]
@marvis  can you please tell what does "warm up training " mean can you please igve a description of what you have done
